File: springboot-version/backend/src/main/java/com/petconnect/entity/MissingPetReport.java
package com.petconnect.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "missing_pet_reports")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MissingPetReport {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "pet_id")
    private Pet pet; // Optional link to an existing pet in the system

    @Column(nullable = false)
    private String petName;

    @Column(nullable = false)
    private String species;

    private String breed;
    private String description;
    private String lastSeenLocation;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    private String imageUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportStatus status; // MISSING, FOUND, REUNITED

    @ManyToOne
    @JoinColumn(name = "reporter_id", nullable = false)
    @org.hibernate.annotations.OnDelete(action = org.hibernate.annotations.OnDeleteAction.CASCADE)
    private User reporter;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    public enum ReportStatus {
        MISSING, FOUND, REUNITED
    }
}

File: aspnet-version/backend/PetConnectAlmost/Models/MissingPetReport.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PetConnectAlmost.Models
{
    [Table("missing_pet_reports")]
    public class MissingPetReport
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }

        [Column("reporter_id")]
        public long ReporterId { get; set; }

        [Column("pet_id")]
        public long? PetId { get; set; }

        [Required]
        [Column("pet_name")]
        public string PetName { get; set; }

        [Required]
        [Column("species")]
        public string Species { get; set; }

        [Column("breed")]
        public string? Breed { get; set; }

        [Column("description")]
        public string? Description { get; set; }

        [Column("last_seen_location")]
        public string? LastSeenLocation { get; set; }
        
        [Column("latitude")]
        public double Latitude { get; set; }
        
        [Column("longitude")]
        public double Longitude { get; set; }

        [Column("image_url")]
        public string? ImageUrl { get; set; }

        [Required]
        [Column("status")]
        public string Status { get; set; } // MISSING, FOUND, REUNITED

        [Column("created_at")]
        public DateTime CreatedAt { get; set; }

        // Navigation
        [ForeignKey("ReporterId")]
        public virtual User Reporter { get; set; }

        [ForeignKey("PetId")]
        public virtual Pet? Pet { get; set; }

        public virtual ICollection<MissingPetContact> Contacts { get; set; } = new List<MissingPetContact>();
    }
}
